Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.py

```diff
@@ -730,29 +730,53 @@ def get_a_list_of_unanalyzed_abuse_mailbox_campaigns_command(client, args):
     return command_results
 
 
-def generate_threat_incidents(threats, current_iso_format_time):
+def generate_threat_incidents(client, threats, current_iso_format_time):
     incidents = []
     for threat in threats:
-        incident = {"dbotMirrorId": str(threat["threatId"]), "name": "Threat", "occurred": current_iso_format_time,
-                    'details': "Threat"}
+        threat_details = None
+        try:
+            threat_details = client.get_details_of_a_threat_request(threat["threatId"])
+        except Exception as e:
+            logging.error(f"Failed to fetch details for threat {threat}: {e}")
+
+        incident = {
+            "dbotMirrorId": str(threat["threatId"]),
+            "name": "Threat",
+            "occurred": current_iso_format_time,
+            "rawJSON": json.dumps(threat_details) if threat_details else {}
+        }
         incidents.append(incident)
+
     return incidents
 
 
-def generate_abuse_campaign_incidents(campaigns, current_iso_format_time):
+def generate_abuse_campaign_incidents(client, campaigns, current_iso_format_time):
     incidents = []
     for campaign in campaigns:
+        campaign_details = None
+        try:
+            campaign_details = client.get_details_of_an_abuse_mailbox_campaign_request(campaign["campaignId"])
+            logging.info(f"this is the campaign details {campaign_details}")
+        except Exception as e:
+            logging.error(f"Failed to fetch details for campaign {campaign}: {e}")
+
         incident = {"dbotMirrorId": str(campaign["campaignId"]), "name": "Abuse Campaign", "occurred": current_iso_format_time,
-                    'details': "Abuse Campaign"}
+                    'details': "Abuse Campaign", "rawJSON": json.dumps(campaign_details) if campaign_details else {}}
         incidents.append(incident)
     return incidents
 
 
-def generate_account_takeover_cases_incidents(cases, current_iso_format_time):
+def generate_account_takeover_cases_incidents(client, cases, current_iso_format_time):
     incidents = []
     for case in cases:
+        case_details = None
+        try:
+            case_details = client.get_details_of_an_abnormal_case_request(case["caseId"])
```

Comment:
Do you not have all the details in the case? Why do we need this request?

Reply:
same as above